From 1e44228becd6f3809c3110936e999ffc3312125b Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Wed, 11 Sep 2024 17:38:06 +0300
Subject: [PATCH] libtool.m4: Avoid warning message in test case 30

The test case for deplibs_check_method in tests/demo.at produces a
warning on multiple systems, such as Cygwin and Mingw. To avoid this
warning when using file_magic, '^x86 archive static' is appended for
lt_cv_deplibs_check_method.

* m4/libtool.m4: Append '^x86 archive static' for file_magic in the
lt_cv_deplibs_check_method.
---
 m4/libtool.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/m4/libtool.m4 b/m4/libtool.m4
index 6a6d13c5..99ebf60f 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -3522,7 +3522,7 @@ bsdi[[45]]*)
 
 cygwin*)
   # func_win32_libid is a shell function defined in ltmain.sh
-  lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
+  lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL|^x86 archive static'
   lt_cv_file_magic_cmd='func_win32_libid'
   ;;
 
@@ -3531,7 +3531,7 @@ mingw* | windows* | pw32*)
   # func_win32_libid shell function, so use a weaker test based on 'objdump',
   # unless we find 'file', for example because we are cross-compiling.
   if ( file / ) >/dev/null 2>&1; then
-    lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
+    lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL|^x86 archive static'
     lt_cv_file_magic_cmd='func_win32_libid'
   else
     # Keep this pattern in sync with the one in func_win32_libid.
-- 
2.45.2

